haskell-indexer-backend-ghc/src/Language/Haskell/Indexer/Backend/Ghc.hs

```diff
@@ -220,8 +220,22 @@ declsFromRenamed ctx (hsGroup, _, _, _) =
                                 $ hsGroup
         explicitVarBindDefs = map mkDeclName explicitNames
         varBindDefs = explicitVarBindDefs ++ implicitVarBindDefs
-    in (concat [defs, instDefs, varBindDefs], instChanges)
+        valBindDefs = case hs_valds hsGroup of
+            ValBinds _ _ _ ->
+                error "should not hit ValBinds when accessing renamed AST"
+            XValBindsLR (NValBinds binds _sigs) -> collectPatSymDefs binds
+    in (concat [defs, instDefs, varBindDefs, valBindDefs], instChanges)
```

Comment:
Could you make valBindDefs and varBindDefs more distinct? I nearly thought it was the same thing twice.
variableDefs and valueDefs seem easier to tell apart.